### PR TITLE
Husky set up

### PR DESCRIPTION
**Pull request by luke-song**
husky set up for pre commit check for linting and cargo fmt done

### CODE REVIEW

**Review by rado0x54 — CHANGES_REQUESTED** [review]
We are using `yarn` and not `npm`, correct. Therefore I'd expect a `yarn.lock` and not a package-lock.json.

File: package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        "npm-run-all": "^4.1.5"
+    },
+    "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1",
+        "prepare": "husky install",
+        "lint": "cd solana/program_v2 && yarn lint"
+    },
+    "husky": {
+        "hooks": {
+        "pre-commit": "cargo fmt --manifest-path solana/program_v2/Cargo.toml --all -- --check && yarn lint"
```
Comment (rado0x54) [inline package.json:20]: This seems duplicate to `.husky/pre-commit` I'm sure that does not need to be necessary?

File: package.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        "doc": "docs"
+    },
+    "devDependencies": {
+        "husky": "^8.0.1",
+        "npm-run-all": "^4.1.5"
+    },
+    "scripts": {
+        "test": "echo \"Error: no test specified\" && exit 1",
+        "prepare": "husky install",
+        "lint": "cd solana/program_v2 && yarn lint"
```
Comment (rado0x54) [inline package.json:16]: I feel the root-level package json should ONLY contain husky related stuff and link out to other projects at only a single point.

e.g. if we want to add the another folder to be linted now how would we add it?

File: package.json (hunk 1 above)
Comment (luke-song) [reply package.json]: yeah, you were right, I deleted this duplication.

File: package.json (hunk 2 above)
Comment (luke-song) [reply package.json]: > We are using `yarn` and not `npm`, correct. Therefore I'd expect a `yarn.lock` and not a package-lock.json.

changed npm to yarn

File: package.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      "pre-commit": ""
+    }
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/identity-com/on-chain-identity-gateway.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
```
Comment (rado0x54) [inline package.json:28]: Can you double check all licenses in the Repo? I think they are all MIT, therefore it should be the same here.

Comment (luke-song) [reply package.json]: fixed

**Comment by luke-song** [comment]
new branch created to clean up the messy commits, and git issues.

**Pull request closed without merging**